### ISSUE TITLE
JavadocStyleCheck should report violation on AST element (line and column)

### ISSUE THREAD

**Issue by Rahulkhinchi03** [checkstyle/checkstyle#11549]
I have read check documentation: https://checkstyle.sourceforge.io/config_javadoc.html#JavadocStyle
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

Found this issue when trying to resolve #8151 

```bash
/var/tmp $ javac SuppressionXpathRegressionJavadocStyle.java

/var/tmp $ cat config.xml

<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocStyle">
        </module>
    </module>
</module>

/var/tmp $ cat SuppressionXpathRegressionJavadocStyle.java

package main;

public class SuppressionXpathRegressionJavadocStyle {
  /**
   * Some description here. // OK
   */
  private void methodWithValidCommentStyle() {}

  /**
   * Some description here // warn
   */
  private void methodWithInvalidCommentStyle() {}
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-X.XX-all.jar -c config.xml SuppressionXpathRegressionJavadocStyle.java


Starting audit...
[ERROR] C:\Users\ASUS\Desktop\opensource\config_checkstyle\src\main\SuppressionXpathRegressionJavadocStyle.java:9: First sentence should end with a period. [JavadocStyle]
Audit done.
Checkstyle ends with 1 error.
```

**Describe what you expect in detail.**

---
The input should not give any violation but still, it gives errors on number 9 and without column number. Need to report a violation on AST element (line and column).

**Comment by nick-mancuso** [comment]
> input should not give any violation

> Need to report a violation

Your expectation is not clear. 

Please clarify your expectation, and add line comments to show where violation is now, and where it should be.

**Comment by Rahulkhinchi03** [comment]
Updated the comment. Please check now.

**Comment by nick-mancuso** [comment]
> There should be no violation in the input file.

Punctuation is part of this check, and the Javadoc starting on line 9 has a description that does not end with a period . So I would expect a violation for the given input.

**Comment by Rahulkhinchi03** [comment]
`java:9: First sentence should end with a period. [JavadocStyle]`

@nick-mancuso It doesn't provide the column number. How can we resolve that?

**Comment by nick-mancuso** [comment]
Please add line comments to your example like this: 

```

/** // violation is here
  * Some description here // but I would expect violation here
 */


```

And remove any expectation of no violations in this file.

**Comment by Rahulkhinchi03** [comment]
Changes Done

**Comment by nick-mancuso** [comment]
Remove line numbers for easy copy/paste.

**Comment by Rahulkhinchi03** [comment]
Changes Done

**Comment by romani** [comment]
here is explanation on why we do not support logging on AST - https://checkstyle.org/config_filters.html#SuppressionXpathFilter

>  the filter does not support suppressions inside javadoc reported by Javadoc checks

**Comment by nihaanth** [comment]
Javadoc comment for methodWithInvalidCommentStyle() should end with a period.

**Comment by romani** [comment]
JavadocStyle is removed from Checkstyle at 13.9.0, users should use SummaryJavadoc instead
Details:  https://github.com/checkstyle/checkstyle/issues/20610
